### PR TITLE
Added INSTR Support for Synapse.

### PR DESCRIPTION
**Pull request by saranshmanglani**

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/sql/dialect/MssqlSqlDialect.java [hunk 1]
```diff
@@ -122,8 +122,10 @@ public MssqlSqlDialect(Context context) {
       case TRIM:
         unparseTrim(writer, call, leftPrec, rightPrec);
         break;
-      case OTHER_FUNCTION:
       case TRUNCATE:
+        unparseTruncateAndRound(writer, call, leftPrec, rightPrec);
```
Comment (roy-sounak) [inline core/src/main/java/org/apache/calcite/sql/dialect/MssqlSqlDialect.java:126]: Revert this change into a seperate PR

File: core/src/main/java/org/apache/calcite/sql/dialect/MssqlSqlDialect.java [hunk 2]
```diff
@@ -178,17 +180,21 @@ public void unparseOtherFunction(SqlWriter writer, SqlCall call, int leftPrec, i
       writer.endFunCall(logFrame);
       break;
     case "ROUND":
-    case "TRUNCATE":
-      final SqlWriter.Frame funcFrame = writer.startFunCall("ROUND");
-      for (SqlNode operand : call.getOperandList()) {
-        writer.sep(",");
-        operand.unparse(writer, leftPrec, rightPrec);
+      unparseTruncateAndRound(writer, call, leftPrec, rightPrec);
```
Comment (roy-sounak) [inline core/src/main/java/org/apache/calcite/sql/dialect/MssqlSqlDialect.java:183]: Revert "truncate and Round" change into a seperate PR

File: core/src/main/java/org/apache/calcite/sql/dialect/MssqlSqlDialect.java [hunk 3]
```diff
@@ -365,6 +371,20 @@ private void unparseTrim(SqlWriter writer, SqlCall call, int leftPrec, int right
     }
   }
 
+  private void unparseTruncateAndRound(
```
Comment (roy-sounak) [inline core/src/main/java/org/apache/calcite/sql/dialect/MssqlSqlDialect.java:374]: Revert "truncate and Round" change into a seperate PR

**Review by roy-sounak — CHANGES_REQUESTED** [review]

**Review by roy-sounak — APPROVED** [review]
LGTM!

**Review by soma-mondal — APPROVED** [review]

**Pull request merged**